Add support for PrimTyConI

TeofilC opened the pull request:
With GHC>8.10 we let the 'reifyType' machinery figure out all the details for us.

On older GHC we are forced to make do with less. We cannot determine the return kind and we cannot determine the kind of the type parameters. So, things get defaulted to 'StarT'.

Resolves #113

I haven't tested with GHC older than 8.8.4 but hopefully the tests pass on CI

RyanGlScott commented on src/Language/Haskell/TH/Datatype.hs:
```diff
@@ -442,7 +442,14 @@ normalizeInfo = normalizeInfo' "normalizeInfo" isn'tReified
   case i of
-    PrimTyConI{}                      -> bad "Primitive type not supported"
+    (PrimTyConI name arity unlifted) -> do
+#if MIN_VERSION_template_haskell(2,16,0)
```
Why is CPP used here? Leave a comment stating why.

RyanGlScott reviewed (commented):
Thanks, @TeofilC! The approach here looks sensible implementation-wise. My only request is to add some more comments:

1. Mention in the Haddocks for `Datatype` that it can be used for primitive data types in addition to orindary `data` declarations.
2. Mention in the Haddocks for `reifyDatatype` that reifying a primitive data type will yield a `DatatypeInfo` without any `datatypeCons`.

TeofilC commented:
Thanks for the review @RyanGlScott . I've pushed a fixup commit addressing your points hopefully.

> Mention in the Haddocks for reifyDatatype that reifying a primitive data type will yield a DatatypeInfo without any datatypeCons.
On this point, I've noted that unboxed sums and tuples are an exception to this rule.

Let me know what you think

RyanGlScott reviewed (approved):
Fantastic. Thanks, @TeofilC!

TeofilC commented:
Thanks!
